tests/test_frontdoor.py

```diff
@@ -304,6 +304,43 @@ def test_back_button():
     assert form["country"] == "England"
 
 
+@pytest.mark.parametrize(
+    ("supplier_name", "expected_text"),
+    [
+        ("Shell", "Shell is now owned by Octopus Energy."),
+        ("Bulb, now part of Octopus Energy", "Bulb is now owned by Octopus Energy."),
+        ("Utility Warehouse", "Referral requests from UW customers will be managed by E.ON Next"),
+    ],
+)
+def test_own_property_back_button_with_supplier_should_return_to_supplier_warning_page(supplier_name, expected_text):
+    client = utils.get_client()
+    page = client.get("/start")
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.status_code == 200
+    session_id = page.path.split("/")[1]
+    assert uuid.UUID(session_id)
+
+    form = page.get_form()
+    form["country"] = "England"
+    page = form.submit().follow()
```

Comment:
seems a good time to use `_check_page`? ie
```suggestion
    page = _check_page(page, "country", "country", "England")
```

Reply:
Also checked the supplier lines below to check page